src/main/java/org/bananarama/crud/sql/SqlDeleteOperation.java

```diff
@@ -70,10 +70,13 @@ public SqlDeleteOperation(DataSource dataSource, Class<T> clazz) {
         String currentTableName = getTableNameForCurrentSession(options);
         
         String whereClauseString = null;
-        if(whereClause instanceof String)
-            whereClauseString = (String) whereClause;
-        else if(whereClause instanceof com.googlecode.cqengine.query.Query)
-            whereClauseString = CQE2SQL.convertCqQuery((com.googlecode.cqengine.query.Query<?>) whereClause);
+        if (whereClause != null) {
+            if(whereClause instanceof String)
+                whereClauseString = (String) whereClause;
+            else if(whereClause instanceof com.googlecode.cqengine.query.Query)
+                whereClauseString = CQE2SQL.convertCqQuery((com.googlecode.cqengine.query.Query<?>) whereClause);            
+        }else
```

Comment:
Initlalize whereClauseString as empty string and leave if clause only. else statement is redundant and should be removed.